rclpy/src/rclpy/node.cpp

```diff
@@ -0,0 +1,77 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Include pybind11 before rclpy_common/handle.h includes Python.h
+#include <pybind11/pybind11.h>
+
+#include <rcl/context.h>
+#include <rcl/error_handling.h>
+#include <rcl/expand_topic_name.h>
+#include <rcl/rcl.h>
+#include <rcl/remap.h>
+#include <rcl/types.h>
+#include <rcl/validate_topic_name.h>
+#include <rcutils/error_handling.h>
+#include <rmw/error_handling.h>
+#include <rmw/validate_full_topic_name.h>
+#include <rmw/validate_namespace.h>
+#include <rmw/validate_node_name.h>
```

Comment:
It seems like there are a lot of unused headers here. I see reason for:

* pybind11/pybind11.h for all the `py::` stuff
* rcl/node.h for `rcl_node_t`, `rcl_node_get_name()`, `rcl_node_get_namespace()`
* rclpy_common/handle.h for the `rclpy_handle_get_pointer_from_capsule

And if you change the code to throw `RCLError` instead: "rclpy_common/exceptions.hpp"

But if you don't change it and instead just want to clear the error state: "rcl/error_handling.h"

I'm pretty sure the rest are unused.